src/platform/tests/TestDnssd.cpp

```diff
@@ -122,13 +166,89 @@ static void DnssdErrorCallback(void * context, CHIP_ERROR error)
     NL_TEST_ASSERT(ctx->mTestSuite, error == CHIP_NO_ERROR);
 }
 
+void TestDnssdBrowse_DnssdInitCallback(void * context, CHIP_ERROR error)
+{
+    auto * ctx = static_cast<DnssdContext *>(context);
+    NL_TEST_ASSERT(ctx->mTestSuite, error == CHIP_NO_ERROR);
+    auto * suite = ctx->mTestSuite;
+
+    NL_TEST_ASSERT(suite,
+                   ChipDnssdBrowse("_mock", DnssdServiceProtocol::kDnssdProtocolUdp, chip::Inet::IPAddressType::kAny,
+                                   chip::Inet::InterfaceId::Null(), HandleBrowse, context,
+                                   &ctx->mBrowseIdentifier) == CHIP_NO_ERROR);
+}
+
+void TestDnssdBrowse(nlTestSuite * inSuite, void * inContext)
```

Comment:
I think this could use some documentation.  Is the difference from `TestDnssdPublishService` that here we are using minmdns, not platform, to publish, but still using the platform browse/resolve?  Except we're not doing the "normal" platform browse/resolve, in that we always do the resolve, right?

Reply:
OK, I'll add a documentation for that.

The key point of this test is to provide a reliable (platform independent) way of setting up mDNS responder. For now the responder responses without additional fields, so e.g. in case of PTR request the responder replies only with PTR record. If I understand correctly the interoperability issue with Apple HomeKit, the case is that there is no AAAA additional record in the response, so client has to request for that in a separate call. So, I thought that this test might ensure that any platform which runs and passes this test should be OK to work with HomeKit.

Reply:
Ah, this will catch browse implementations that actually fail out on internally if the PTR lookup does not provide a AAAA record, is the idea?

Reply:
Yes, that's the idea. However, in the DNS-SD thread on slack, I've seen that there are also other possibilities of failure. E.g. responder will provide additional records but not all of them, so implementation might check only for one thing, e.g. only for "A" record, in order to request for all missing things. So, in case where "A" is provided but AAAA is not, the implementation will fail as well. In order to verify that, we will need all permutations of possible combinations which might not be ideal for a unit test.... But this simple test case will at least check for a simple case.

>  will also catch things that assume an SRV query will include the AAAA as well

yes, this will be covered here